services/debugger/destination/eventDeliveryStatusUploader.go

```diff
@@ -35,23 +35,43 @@ var uploader debugger.UploaderI
 var (
 	configBackendURL                  string
 	disableEventDeliveryStatusUploads bool
+	eventsDeliveryCacheMap            map[string][]*DeliveryStatusT
+	eventsCacheMapLock                sync.RWMutex
```

Comment:
Would it be possible to avoid using global variables and instead have them as struct properties. Maybe `EventDeliveryStatusUploader` struct would be suitable for this.

Reply:
Sounds good.